src/viam/sdk/components/camera.hpp

```diff
@@ -65,6 +65,9 @@ class Camera : public Component {
 
         /// @brief Contains the mime types the camera supports.
         Camera::mime_types mime_types;
+
+        /// @brief Contains the camera's frame rate.
+        float frame_rate;
```

Comment:
I think we probably want to use `double` here instead of `float`

Reply:
I used a float32 in the api changes so that is why I put in float here, but good with me! just for my own understanding why double instead of float here?

Reply:
Actually sorry that's my mistake, thought this was a situation where a data format was using `float` to mean "this number has decimals" but we should translate it to `double` in C++, but I'm now seeing there are other components which do in fact use `float`

Reply:
got it - in general is it better to use doubles instead of floats in C++?

Reply:
You can think of it as the default "thing with a decimal" yeah, in some sense this is what the C++ language does too because a number with a decimal is deduced as double by default:
```cpp
template<class T>
void f(T t);
void g() { 
    f(5.0); // calls f<double>
}

auto x = 5.0; // x has type double
```
As for which is better, I think from a clarity/usability perspective it's safe to go with `double` by default, but if you're in an application or environment doing serious number crunching or with memory/bandwidth constraints you can decide if `float` might make more sense

Reply:
I see. thank you for explaining!